collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -731,25 +745,94 @@ public void recordBean(
                     matchedRule.labelNames,
                     matchedRule.labelValues,
                     value.doubleValue());
-            addSample(
-                    new MetricFamilySamples.Sample(
-                            matchedRule.name,
-                            matchedRule.labelNames,
-                            matchedRule.labelValues,
-                            value.doubleValue()),
-                    matchedRule.type,
-                    matchedRule.help);
+
+            final MatchedRule finalMatchedRule = matchedRule;
+
+            switch (matchedRule.type) {
+                case "COUNTER":
+                    {
+                        CounterSnapshot.Builder counterBuilder =
+                                countersMap.computeIfAbsent(
+                                        matchedRule.name,
+                                        name ->
+                                                CounterSnapshot.builder()
+                                                        .name(
+                                                                PrometheusNaming.sanitizeMetricName(
```

Comment:
The `sanitizeMetricName()` calls here are not needed because there is a previous call to `JmxCollector.safeName()`. However, we might refactor `JmxCollector.safeName()` to something like this:
```java
    static String safeName(String name) {
        if (name == null) {
            return null;
        }
        String sanitizedName = PrometheusNaming.sanitizeMetricName(name);
        while (sanitizedName.contains("__")) {
            sanitizedName = sanitizedName.replace("__", "_");
        }
        return sanitizedName;
    }
```

Reply:
In the process of testing this change and determined that `PrometheusNaming.sanitizeMetricName` doesn't handle two scenarios defined in `SafeNameTest`

An empty string:

- `JmxCollector.safeName()` returns an empty string
- `PrometheusNaming.sanitizeMetricName` throws an `IllegalArgumentException`

I believe throwing an `IllegalArgumentException` is the correct behavior. Please confirm.

A metric prefixed with a number:

- `JmxCollector.safeName()` returns a String prefixed with `_`
- `PrometheusNaming.sanitizeMetricName` returns the string unmodified

Can you confirm the correct behavior?

Reply:
* I agree throwing a an `IllegalArgumentException` is the right behavior if the metric name is empty.
* If `sanitizeMetricName()` returns metric names starting with a digit that's a bug, I'll fix it in `client_java`.